src/sugarblas.jl

```diff
@@ -17,25 +16,24 @@ import Base.LinAlg.BLAS: blascopy!, scal!, scal
     DY
 end
 
-# Add robustness to scal!
-@inbounds function scal!(n::Integer, DA::Number, DX::AbstractArray, incx::Integer)
-    i = 1
-    n = min(n,length(DX))
-    @simd for i in 1:incx:n
-        DX[i] *= DA
-    end
-    DX
+function scal(n::Integer, DA::Number, DX::AbstractArray, incx::Integer)
+    O = copy(DX)
+    scal!(n, DA, O, incx)
 end
 
-# Add robustness to scal
-@inbounds function scal{T}(n::Integer, DA::Number, DX::AbstractArray{T}, incx::Integer)
-    zeros(T, size(DX)...)
+@inbounds function scal!(n::Integer, DA::Number, DX::AbstractArray, incx::Integer)
     i = 1
     n = min(n,length(DX))
-    @simd for i in 1:incx:n
-        O[i] = DA*DX[i]
+    if incx == 1
+        @simd for i in 1:n
+            DX[i] *= DA
```

Comment:
According to [this](http://docs.julialang.org/en/release-0.4/manual/performance-tips/#performance-annotations) both `@fastmath` and `@inbounds` can.

> Note: While @simd needs to be placed directly in front of a loop, both @inbounds and @fastmath can be applied to several statements at once, e.g. using begin ... end, or even to a whole function.